src/ice.rs

```diff
@@ -1,13 +1,21 @@
 use std::error::Error;
 
+macro_rules! ice {
+    ($($tt:tt)*) => {
+        panic!("ICE {}", format_args!($($tt)*))
+    }
+}
+
 pub trait Ice<T> {
     fn ice(self, message: &str) -> T;
```

Comment:
This really should be called `unwrap_or_ice()`.

Reply:
fixed